components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/odata/RDBMSDataHandler.java

```diff
@@ -1016,6 +1016,10 @@ private String getBase64StringFromInputStream(InputStream in) throws SQLExceptio
         try {
             if (meta.getDatabaseProductName().toLowerCase().contains(ORACLE_SERVER)) {
                 resultSet = meta.getColumns(null, meta.getUserName(), tableName, null);
+                if(!meta.getConnection().getSchema().equals(meta.getUserName())){
```

Comment:
Shall we do a null check too for meta.getConnection().getSchema(). The best way is to switch meta.getUserName() with meta.getConnection().getSchema() and do a null check on meta.getUserName(), because meta cannot be null

Reply:
ok. I will look into this now and see.

Reply:
I did add some null checks. Is it ok?